dateutil/parser/_parser.py

```diff
@@ -1202,7 +1202,12 @@ def _assign_tzname(self, dt, tzname):
 
     def _to_decimal(self, val):
         try:
-            return Decimal(val)
+            decimal_value = Decimal(val)
+            # See GH 662, edge case, infinite value should not be converted via `_to_decimal`
+            if not decimal_value.is_finite():
+                raise ValueError("Converted decimal value is infinite or NaN")
+
+            return decimal_value
```

Comment:
I think it's preferable to put the `return` statement outside the `try`/`except` block.

Reply:
Sounds good.